src/config/scrivitoSites.ts

```diff
@@ -42,7 +42,7 @@ export async function ensureSiteIsPresent() {
       const preferredLanguageOrder = [...window.navigator.languages, 'en', null]
 
       for (const language of preferredLanguageOrder) {
-        const site = sites.find((site) => siteHasLanguage(site, language))
+        const site = sites.find((site) => siteIsLanguageVersion(site, language))
```

Comment:
Does not seem to work properly. E.g. in this working copy: https://edit.scrivito.com/d0a154d76edf2a7bd991fc658e700a1d~https://dave-ensure-valid-site.scrivito-portal-app.pages.dev/?_scrivito_workspace_id=v358d491ab7aace5&_scrivito_display_mode=view

There is maling-app content present. I don't see a redirect take place.